Source/Core/Gen4/Encounters4.cpp

```diff
@@ -272,6 +272,125 @@ namespace Encounters4
                 }
             }
 
+            if (encounter == Encounter::SafariZone)
+            {
+                size_t size = hgss_safari.size();
+
+                for (size_t offset = 0; offset < size; )
+                {
+                    const u8 *entry = hgss_safari.data() + offset;
+                    int grassBlockSlot = 0;
+                    int waterBlockSlot = 0;
+
+                    u8 location = entry[0];
+                    u8 waterFlag = entry[1];
+
+                    std::vector<Slot> slots;
+                    for (int i = 0; i < 10; i++)
+                    {
+                        u16 specie = *reinterpret_cast<const u16 *>(entry + (30 * modifier) + 2 + (i * 3));
+                        u8 level = entry[(30 * modifier) + 4 + (i * 3)];
+
+                        while (grassBlockSlot < 10)
```

Comment:
for (int grassBlockSlot = 0; grassBlockSlot  < 10; grassBlockSlot) {

}

Reply:
This makes things not working